src/threechances.cpp

```diff
@@ -137,81 +160,107 @@ int findKeyDown(std::map<int, bool> *keysPressed) {
-				keysPressed[LEFT] = true;
-				lastKeyPressed = LEFT;
-				endPoint = level.getX() + TILE_SIZE * SCALE;
-				player.moveInDirection(levelGrid, monsterGrid, LEFT, endPoint);
+	GENERAL_STATE gs = gameControl->getGeneralState();
+
+	switch (gs) {
+		case GENERAL_STATE::menu: {
+			// detect space
+			if (input->isKeyDown(SPACE_KEY)) {
+				gameControl->setGeneralState(GENERAL_STATE::game);
 			}
 
-			if (input->isKeyDown(RIGHT_KEY) && !keysPressed[RIGHT]) {
-				keysPressed[RIGHT] = true;
-				lastKeyPressed = RIGHT;
-				endPoint = level.getX() - TILE_SIZE * SCALE;
-				player.moveInDirection(levelGrid, monsterGrid, RIGHT, endPoint);
+		} break;
+		case GENERAL_STATE::paused: {
+			if (input->isKeyDown(SPACE_KEY)) {
+				gameControl->setGeneralState(GENERAL_STATE::game);
+			}
+		} break;
+		case GENERAL_STATE::game: {
+			if (input->isKeyDown(ESC_KEY)) {
+				gameControl->setGeneralState(GENERAL_STATE::paused);
 			}
 
-			if (input->isKeyDown(UP_KEY) && !keysPressed[UP]) {
-				keysPressed[UP] = true;
-				lastKeyPressed = UP;
-				endPoint = level.getY() + TILE_SIZE * SCALE;
-				player.moveInDirection(levelGrid, monsterGrid, UP, endPoint);
+			// Check no animation currently running
+			if (!player.getAnimating()) {
+				// Check if it's player's turn
+				if (gameControl->getGameState() == GAME_STATE::player) {
+					float endPoint;
+
+					if (input->isKeyDown(LEFT_KEY) && !keysPressed[LEFT]) {
+						keysPressed[LEFT] = true;
+						lastKeyPressed = LEFT;
+						endPoint = level.getX() + TILE_SIZE * SCALE;
+						player.moveInDirection(levelGrid, monsterGrid, LEFT, endPoint, gameControl);
+					}
+
+					if (input->isKeyDown(RIGHT_KEY) && !keysPressed[RIGHT]) {
+						keysPressed[RIGHT] = true;
+						lastKeyPressed = RIGHT;
+						endPoint = level.getX() - TILE_SIZE * SCALE;
+						player.moveInDirection(levelGrid, monsterGrid, RIGHT, endPoint, gameControl);
+					}
+
+					if (input->isKeyDown(UP_KEY) && !keysPressed[UP]) {
+						keysPressed[UP] = true;
+						lastKeyPressed = UP;
+						endPoint = level.getY() + TILE_SIZE * SCALE;
+						player.moveInDirection(levelGrid, monsterGrid, UP, endPoint, gameControl);
+					}
+
+					if (input->isKeyDown(DOWN_KEY) && !keysPressed[DOWN]) {
+						keysPressed[DOWN] = true;
+						lastKeyPressed = DOWN;
+						endPoint = level.getY() - TILE_SIZE * SCALE;
+						player.moveInDirection(levelGrid, monsterGrid, DOWN, endPoint, gameControl);
+					}
+				}
+				// Enemy's turn
+				else {
+					enemyAi();
+				}
 			}
+			// Animates sprites for player turn
+			else {
+				int oppDirection = -1;
+				// player direction is set in player.cpp
+				if (player.getDirection() == LEFT)
+					oppDirection = RIGHT;
+				else if (player.getDirection() == RIGHT)
+					oppDirection = LEFT;
+				else if (player.getDirection() == UP)
+					oppDirection = DOWN;
+				else if (player.getDirection() == DOWN)
+					oppDirection = UP;
+
+				if (player.getDirection() != ATTACK) {
```

Comment:
Can we replace the `direction` variable in player class with `action` instead? Like what i've done in the ghost class